src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -686,55 +888,53 @@ export default class CanvasController {
 
                 cc.stroke();
 
-                if (exit_name) {
+                if (exitName) {
                     // Initialize count for this transition
-                    if (isNaN(text_at_point[exit_name])) {
-                        text_at_point[exit_name] = 0;
+                    if (isNaN(textAtPoint[exitName])) {
+                        textAtPoint[exitName] = 0;
                     }
 
                     // Move the y point for drawing depending on how many sids we have drawn text for
                     // at this point already
-                    const y_point = fixY + (15 * text_at_point[exit_name]);
-                    cc.fillText(`${sid.identifier}.${exit_name}`, fixX + 10, y_point);
+                    const y_point = fixY + (15 * textAtPoint[exitName]);
+                    cc.fillText(`${sid.identifier}.${exitName}`, fixX + 10, y_point);
 
-                    text_at_point[exit_name] += 1;  // Increment the count for this transition
+                    textAtPoint[exitName] += 1;  // Increment the count for this transition
                 }
-            });
+            }
 
-            if (write_sid_name) {
-                cc.fillText(sid.identifier, fixX + 10, fixY);
+            if (shouldDrawProcedureName) {
+                const labelOffsetX = fixX + 10;
+                cc.fillText(sid.identifier, labelOffsetX, fixY);
             }
-        });
+        }
     }
 
     /**
      * Draw a trailing indicator 2.5 NM (4.6km) behind landing aircraft to help with traffic spacing
      *
      * @for CanvasController
      * @method canvas_draw_separation_indicator
-     * @param cc
-     * @param aircraft
+     * @param cc {HTMLCanvasContext}
+     * @param aircraft {AircraftModel}
      */
     canvas_draw_separation_indicator(cc, aircraft) {
-        if (!GameController.shouldUseTrailingSeparationIndicator(aircraft)) {
+        if (!GameController.shouldUseTrailingSeparationIndicator(aircraft) || !this.theme.RADAR_TARGET.TRAILING_SEPARATION_INDICATOR_ENABLED) {
```

Comment:
This is already be included in the `.shouldUseTrailingSeparationIndicator()` method, so there's no need for the addition here.

Reply:
This block is mostly as I found it, I just removed an unnecessary parenthesis. This will be simplified in a future PR, I just didn't want to add more to the pile for this PR.

Reply:
right, not the block i'm concerned about, just the if statement. you added an or that is redundant. it should read what it was before:
`if (!GameController.shouldUseTrailingSeparationIndicator(aircraft)) {`

Reply:
and I don't like this block, either.  the sooner we can get rid of it (as confusing as it is) the better!

Reply:
seems my espresso hasn't quite kicked in yet, I was thinking this was an entirely different block.

**_this_** block is actually a combination of two if statements.  Isn't it possible for the theme to specify yes and the `GameController` no?

Reply:
I know there was a second if statement below this that was consolidated into the line being commented on. What I'm trying to say is the if statement that was moved should have been _REmoved_ altogether. It is already being handled in the `.shouldUseTrailingSeparationIndicator()` method:

```
/**
 * Check whether or not the trailing distance separator should be drawn.
 *
 * Used by the `CanvasController` to determine whether or not to proceed with
 * `canvas_draw_separation_indicator`.
 *
 * @for GameController
 * @method shouldUseTrailingSeparationIndicator
 * @param aircraft {AircraftModel}
 * @return {boolean}
 */
shouldUseTrailingSeparationIndicator(aircraft) {
    const userSettingsValue = this.getGameOption(GAME_OPTION_NAMES.DRAW_ILS_DISTANCE_SEPARATOR);
    let isIndicatorEnabled = userSettingsValue === 'yes';

    if (userSettingsValue === 'from-theme') {
        isIndicatorEnabled = this.theme.RADAR_TARGET.TRAILING_SEPARATION_INDICATOR_ENABLED;
    }

    return isIndicatorEnabled && aircraft.isArrival();
}
```

Reply:
marvelous.  I'll make that change!

Reply:
done